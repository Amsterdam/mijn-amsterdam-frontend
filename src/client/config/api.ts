import {
  ApiResponse_DEPRECATED,
  FailedDependencies,
} from '../../universal/helpers/api';
import { ApiError, AppState } from '../../universal/types/App.types';
import { regelingenTitle } from '../pages/Thema/HLI/HLI-thema-config';

export const BFF_API_BASE_URL = import.meta.env.REACT_APP_BFF_API_URL;
export const BFF_API_HEALTH_URL = `${BFF_API_BASE_URL}/status/health`;

export const BFFApiUrls = {
  MAP_DATASETS_WMS: `${BFF_API_BASE_URL}/map/datasets/wms`,
  MAP_DATASETS: `${BFF_API_BASE_URL}/map/datasets`,
  SEARCH_CONFIGURATION: `${BFF_API_BASE_URL}/services/search-config`,
  SERVICES_CMS_MAINTENANCE_NOTIFICATIONS_URL: `${BFF_API_BASE_URL}/services/cms/maintenance-notifications`,
  SERVICES_SAURON: `${BFF_API_BASE_URL}/services/all`,
  SERVICES_SSE: `${BFF_API_BASE_URL}/services/stream`,
  ERFPACHT_DOSSIER_DETAILS: `${BFF_API_BASE_URL}/services/erfpacht/dossier`,
  AFIS_BUSINESSPARTNER: `${BFF_API_BASE_URL}/services/afis/businesspartner`,
  AFIS_FACTUREN: `${BFF_API_BASE_URL}/services/afis/facturen`,
};

// Urls directly used from front-end
export const LOGIN_URL_DIGID = `${BFF_API_BASE_URL}/auth/digid/login`;
export const LOGIN_URL_EHERKENNING = `${BFF_API_BASE_URL}/auth/eherkenning/login`;
export const LOGOUT_URL = `${BFF_API_BASE_URL}/auth/logout`;
export const AUTH_API_URL = `${BFF_API_BASE_URL}/auth/check`;
export const AUTH_API_URL_EHERKENNING = `${BFF_API_BASE_URL}/auth/eherkenning/check`;
export const AUTH_API_URL_DIGID = `${BFF_API_BASE_URL}/auth/digid/check`;

export const loginUrlByAuthMethod: Record<string, string> = {
  eherkenning: LOGIN_URL_EHERKENNING,
  digid: LOGIN_URL_DIGID,
};

export const ExcludePageViewTrackingUrls = [
  LOGIN_URL_DIGID,
  LOGIN_URL_EHERKENNING,
];

/**
 * @deprecated
 */
export const ErrorNames: Record<string /* ApiStateKey */, string> = {
  BRP: 'Persoonlijke gegevens, paspoort, ID-kaart',
  KVK: 'Mijn onderneming',
  JEUGD: 'Jeugd: aanvragen en voorzieningen',
  AFIS: 'Facturen en betalen',
  AFIS_facturenoverview: 'Facturen en betalen: Overzicht van facturen',
  AFIS_afgehandeld: 'Facturen en betalen: Afgehandelde facturen',
  AFIS_open: 'Facturen en betalen: Openstaande facturen',
  AFIS_overgedragen:
    'Facturen en betalen: Facturen in het incasso- en invorderingstraject van directie Belastingen',
  AFVAL: 'Afvalgegevens rond uw adres',
  AFVALPUNTEN: 'Afvalpunten',
  ALL: 'Alle gegevens', // indien data helemaal niet opgehaald kan worden
  AVG: 'Ingediende AVG verzoeken',
  BELASTINGEN: 'Actuele updates over uw belastingen',
  BEZWAREN: 'Ingediende bezwaren',
  BODEM: 'Bodem: loodmetingen',

  CMS_CONTENT: 'Uitleg Mijn Amsterdam',
  ERFPACHT: 'Erfpacht',
  HLI_regelingen: regelingenTitle,
  HLI_stadspas: 'Stadspas, saldo en transacties',
  HORECA: 'Horeca vergunningen',
  KLACHTEN: 'Ingediende klachten',
  KLANT_CONTACT: 'Contactmomenten',
  KREFIA: 'Kredietbank & FIBU',

  MILIEUZONE: 'Milieuzone',
  MY_LOCATION: 'Uw locatie op de kaart',
  NOTIFICATIONS: 'Actuele updates',
  OVERTREDINGEN: 'Overtredingen voertuigen',
  PARKEREN: 'Parkeren',
  PARKEREN_vergunningen: 'Parkeervergunningen',
  SUBSIDIES: 'Subsidies',
  SVWI: 'Werk & Inkomen portaal',
  TOERISTISCHE_VERHUUR_bbVergunningen: 'Uw vergunning Bed & Breakfast',
  TOERISTISCHE_VERHUUR_lvvRegistraties:
    'Toeristische verhuur: Registratienummers + meldingen',
  TOERISTISCHE_VERHUUR_vakantieverhuurVergunningen:
    'Uw vergunning vakantieverhuur',
  TOERISTISCHE_VERHUUR: 'Toeristische verhuur + meldingen',
  VAREN: 'Passagiersvaart',

  VERGUNNINGEN: 'Vergunningen en ontheffingen',
  WMO: 'Zorg en ondersteuning',
  WPI_AANVRAGEN: 'Uitkeringaanvragen',
  WPI_BBZ: 'Aanvraag Bbz',
  WPI_SPECIFICATIES: 'Uitkeringsspecificaties en jaaropgaven',
  WPI_TONK: 'Aanvraag TONK',
  WPI_TOZO: 'Aanvraag Tozo',
};

export function createErrorDisplayData(
  stateKey: string,
  apiResponseData: ApiResponse_DEPRECATED<unknown> | null | string
): ApiError {
  const name = ErrorNames[stateKey] || stateKey;
  const errorMessage =
    (typeof apiResponseData === 'object' &&
    apiResponseData !== null &&
    'message' in apiResponseData
      ? apiResponseData.message
      : null) || 'Communicatie met api mislukt.';
  return {
    stateKey,
    name,
    error: errorMessage,
  };
}

export function createFailedDependenciesError(
  stateKey: string,
  failedDependencies: FailedDependencies
) {
  const apiErrors = [];
  for (const [stateDependencyKey, apiDependencyResponseData] of Object.entries(
    failedDependencies
  )) {
    apiErrors.push(
      createErrorDisplayData(
        `${stateKey}_${stateDependencyKey}`,
        apiDependencyResponseData as ApiResponse_DEPRECATED<unknown>
      )
    );
  }
  return apiErrors;
}

export function getApiErrors(appState: AppState): ApiError[] {
  if (appState) {
    const filteredResponses = Object.entries(appState).filter(
      ([k, apiResponseData]: [string, unknown]) => {
        return (
          !['setAppState', 'isReady', 'setIsAppStateReady'].includes(k) &&
          (typeof apiResponseData !== 'object' ||
            apiResponseData == null ||
            ('status' in apiResponseData &&
              (apiResponseData?.status === 'ERROR' ||
                apiResponseData?.status === 'DEPENDENCY_ERROR' ||
                (apiResponseData?.status === 'OK' &&
                  'failedDependencies' in apiResponseData &&
                  !!apiResponseData?.failedDependencies))))
        );
      }
    );

    const apiErrors = [];

    for (const [stateKey, apiResponseData] of filteredResponses) {
      if (
        apiResponseData?.status === 'OK' &&
        'failedDependencies' in apiResponseData &&
        apiResponseData?.failedDependencies
      ) {
        apiErrors.push(
          ...createFailedDependenciesError(
            stateKey,
            apiResponseData.failedDependencies
          )
        );
      } else {
        apiErrors.push(
          createErrorDisplayData(
            stateKey,
            apiResponseData as ApiResponse_DEPRECATED<unknown>
          )
        );
      }
    }

    return apiErrors;
  }

  return [];
}
