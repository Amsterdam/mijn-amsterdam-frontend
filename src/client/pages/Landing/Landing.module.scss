@import '../../styles/_global.scss';

.LoginBtn {
  display: inline-flex;
  height: 5rem;
  width: 28rem; // ie11 fallback
  width: fit-content;
  text-decoration: none;
  align-items: center;
  white-space: nowrap;

  .LoginButtonText {
    background-color: $color-primary-red;
    color: $color-white;
    font-weight: 700;
    padding-right: 1.3rem;
    padding-left: 1.3rem;
    line-height: 5rem;
  }

  + p {
    margin-top: 1rem;
  }
}

.LoginBtn--eherkenning {
  .LoginButtonText {
    background-color: $color-support-darkblue;
  }
}

.LoginBtnDisabled {
  cursor: progress;
  span {
    background-color: $color-neutral-grey2 !important;
    border-color: $color-neutral-grey2 !important;
    cursor: not-allowed;
    pointer-events: none;
    color: $color-neutral-grey3 !important;
  }
  & {
    filter: grayscale(100%);
  }
}

.LandingContent {
  > p:last-of-type {
    margin-bottom: 9rem;
  }
}

.LoginLogo {
  width: 5rem;
}

.LoginHeading {
  padding-top: 2.6rem;
  padding-bottom: 1.2rem;
}

.LoginOption {
  border-top: 0.1rem solid $color-neutral-grey2;
  padding-top: 2.6rem;
  padding-bottom: 1rem;

  p:last-of-type {
    margin-top: 0;
  }

  &--irma .LoginLogoWrap {
    background-color: transparent;
  }

  &--eherkenning {
    margin-bottom: 3rem;

    .LoginLogoWrap {
      background-color: transparent;
    }
  }
}

.LoginOptionHeading {
  margin-bottom: 1rem;
}

.LoginLogoWrap {
  display: inline-flex;
  align-items: center;
  height: 5rem;
  width: 5rem;
  margin-right: 1rem;
  background-color: #000;
}

.FaqInfo {
  margin: 0;
}

.Heading {
  @include IE11 {
    overflow: hidden;
  }
}
