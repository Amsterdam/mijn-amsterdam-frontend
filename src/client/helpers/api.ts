import { generatePath } from 'react-router-dom';

import { BFFApiUrls } from '../config/api';

export function generateBffApiUrl(
  route: keyof typeof BFFApiUrls,
  routeParams?: Record<string, string>
) {
  const url = new URL(BFFApiUrls[route]);
  return `${url.origin}${generatePath(url.pathname, routeParams)}`;
}

export function generateBffApiUrlWithEncryptedPayloadQuery(
  route: keyof typeof BFFApiUrls,
  encryptedPayload: string,
  routeParams?: Record<string, string>
) {
  return `${generateBffApiUrl(route, routeParams)}?payload=${encodeURIComponent(encryptedPayload)}`;
}
