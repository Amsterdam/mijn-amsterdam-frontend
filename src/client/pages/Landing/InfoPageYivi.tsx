import { AppRoutes } from '../../../universal/config';
import {
  Heading,
  LinkdInline,
  PageContent,
  PageHeading,
  TextPage,
} from '../../components';
import styles from './Landing.module.scss';

export default function InfoPageYivi() {
  return (
    <TextPage>
      <PageHeading className={styles.Heading}>
        Melding openbare ruimte volgen via Mijn Amsterdam
      </PageHeading>
      <PageContent className={styles.LandingContent} id="skip-to-id-AppContent">
        <p>
          Vanaf 1 mei kunt u uw meldingen openbare ruimte ook volgen via Mijn
          Amsterdam, uw persoonlijke online pagina van de gemeente Amsterdam.
          Dit is een proef. Om uw meldingen te zien kunt u{' '}
          <LinkdInline href={AppRoutes.YIVI_LANDING}>
            inloggen met Yivi
          </LinkdInline>
          .
        </p>
        <Heading size="large" el="h2">
          Welke meldingen kan ik volgen?
        </Heading>
        <p>
          Tijdens de proefperiode is het mogelijk om uw meldingen van de
          afgelopen 12 maanden te volgen binnen Mijn Amsterdam. U kunt alleen
          meldingen volgen waarbij u aangeeft dat u op de hoogte wilt blijven en
          uw e-mailadres opgeeft. Het e-mailadres bij uw melding moet hetzelfde
          zijn als het e-mailadres dat u deelt tijdens het inloggen met Yivi.
          Lees hieronder hoe het werkt.
        </p>
        <Heading size="medium" el="h3">
          Hoe werkt het?
        </Heading>
        <p>
          U kunt uw meldingen over de openbare ruimte volgen als u inlogt met
          Yivi. Met Yivi kunt u bewijzen dat u toegang heeft tot het e-mailadres
          dat u heeft opgegeven tijdens het doen van een melding.
        </p>
        <p>Als u nog geen Yivi hebt:</p>
        <ol>
          <li>
            <LinkdInline external href="https://yivi.app">
              Download Yivi
            </LinkdInline>
            .
          </li>
          <li>
            Kies een pincode en vul desgewenst uw e-mailadres in als
            beveiliging.
          </li>
          <li>
            Zodra u bent ingelogd: Voeg uw e-mailadres toe (het e-mailadres dat
            u opgeeft bij het doen van een melding).
          </li>
        </ol>
        <p>Meldingen volgen via Mijn Amsterdam:.</p>
        <ol>
          <li>
            Ga naar{' '}
            <LinkdInline href={AppRoutes.YIVI_LANDING}>
              inloggen met Yivi
            </LinkdInline>
            .
          </li>
          <li>
            Deel uw e-mailadres met de Yivi-app. Deel het e-mailadres dat u bij
            uw melding(en) gebruikt.
          </li>
          <li>
            Als u bent ingelogd, ziet u uw meldingen en de laatste updates.
          </li>
        </ol>
        <Heading size="medium" el="h3">
          Wat is Yivi?
        </Heading>
        <p>
          Yivi is een app waarmee u kunt bewijzen wie u bent, zonder dat u
          onnodige gegevens over u zelf deelt. U ziet altijd wat een organisatie
          van u wil weten. U hebt dus controle over uw gegevens.
          <LinkdInline
            external
            href="https://www.amsterdam.nl/innovatie/digitalisering-technologie/digitalisering/yivi-nieuwe-manier-inloggen/"
          >
            Meer informatie leest u op Yivi: een nieuwe manier van inloggen
          </LinkdInline>
          .
        </p>
        <Heading size="medium" el="h4">
          Na de proefperiode
        </Heading>
        <p>
          De proef loopt tot en met september 2023. Mede op basis van uw mening
          bepaalt de gemeente of we de proefperiode verlengen en/of Yivi op meer
          plekken aanbieden.
        </p>
        <Heading size="medium" el="h4">
          Waarom deze proef
        </Heading>
        <p>
          De gemeente doet deze proef om Amsterdammers kennis te laten maken met
          een digitale identiteit. De gemeente kiest voor Yivi omdat deze app de
          uitgangspunten van de gemeente ondersteunt: Privacy by design, regie
          op eigen gegevens (decentraal opslaan van gegevens), open source,
          dataminimalisatie en alleen data delen die nodig is.
        </p>
        <Heading size="medium" el="h4">
          Geef uw mening
        </Heading>
        <p>
          Geef uw mening via de knop ‘Uw mening’ die u rechts in beeld ziet. Wij
          zijn benieuwd naar uw mening om met Yivi uw meldingen openbare ruimte
          op Mijn Amsterdam te volgen.
        </p>
      </PageContent>
    </TextPage>
  );
}
