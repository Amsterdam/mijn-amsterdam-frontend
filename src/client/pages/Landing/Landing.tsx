import DigiDLogo from '../../assets/images/digid-logo.svg';
import EherkenningLogo from '../../assets/images/eherkenning-logo.svg';
import IrmaLogo from '../../assets/images/irma_logo.jpg';
import classnames from 'classnames';
import { useEffect, useRef, useState } from 'react';
import { FeatureToggle } from '../../../universal/config';
import {
  Heading,
  LinkdInline,
  PageContent,
  PageHeading,
  TextPage,
  Alert,
} from '../../components';
import {
  LOGIN_URL_DIGID,
  LOGIN_URL_EHERKENNING,
  LOGIN_URL_IRMA,
} from '../../config/api';
import { trackPageView } from '../../hooks';
import styles from './Landing.module.scss';
import { ExternalUrls } from '../../config/app';
import { MaintenanceNotification01 } from '../../config/staticData';
import { IS_PRODUCTION } from '../../../universal/config/env';

export default function Landing() {
  const loginButton = useRef(null);

  useEffect(() => {
    trackPageView(
      'Landingspagina',
      document.location.pathname + '/landingspagina'
    );
  }, []);

  const [isRedirecting, setRedirecting] = useState(false);
  const [isRedirectingEherkenning, setRedirectingEherkenning] = useState(false);

  return (
    <TextPage>
      <PageHeading className={styles.Heading}>
        Welkom op Mijn Amsterdam
      </PageHeading>
      <PageContent className={styles.LandingContent} id="skip-to-id-AppContent">
        <p>
          Mijn Amsterdam is uw persoonlijke online pagina bij de gemeente
          Amsterdam.
        </p>
        {new Date() < new Date('2020-09-22T12:00:00') && (
          <Alert type="warning">
            <p>{MaintenanceNotification01.description}</p>
          </Alert>
        )}
        <div className={styles.LoginOption}>
          {FeatureToggle.eherkenningActive && (
            <Heading className={styles.LoginOptionHeading} size="tiny" el="h3">
              Voor particulieren en eenmanszaken
            </Heading>
          )}
          <p>
            <a
              ref={loginButton}
              role="button"
              href={LOGIN_URL_DIGID}
              onClick={() => setRedirecting(true)}
              rel="noopener noreferrer"
              className={classnames(
                styles.LoginBtn,
                (isRedirecting || isRedirectingEherkenning) &&
                  styles.LoginBtnDisabled
              )}
            >
              <span className={styles.LoginLogoWrap}>
                <img
                  src={DigiDLogo}
                  alt="DigiD logo"
                  className={styles.LoginLogo}
                />
              </span>
              <span className={styles.LoginButtonText}>
                {isRedirecting ? 'Bezig met inloggen...' : 'Inloggen met DigiD'}
              </span>
            </a>
          </p>
          <Heading size="tiny" el="h4">
            Hebt u nog geen DigiD? Regel dit dan eerst.
          </Heading>
          <p>
            Ga naar{' '}
            <a rel="noopener noreferrer" href="https://www.digid.nl/aanvragen">
              DigiD aanvragen
            </a>
          </p>
        </div>
        {FeatureToggle.irmaActive && (
          <div
            className={classnames(
              styles.LoginOption,
              styles['LoginOption--irma']
            )}
          >
            <Heading className={styles.LoginOptionHeading} size="tiny" el="h3">
              Voor particulieren
            </Heading>
            <p>
              <a
                ref={loginButton}
                role="button"
                href={LOGIN_URL_IRMA}
                onClick={() => setRedirectingEherkenning(true)}
                rel="noopener noreferrer"
                className={classnames(
                  styles.LoginBtn,
                  styles['LoginBtn--irma'],
                  (isRedirecting || isRedirectingEherkenning) &&
                    styles.LoginBtnDisabled
                )}
              >
                <span className={styles.LoginLogoWrap}>
                  <img
                    src={IrmaLogo}
                    alt="IRMA logo"
                    className={styles.LoginLogo}
                  />
                </span>
                <span className={styles.LoginButtonText}>
                  {isRedirectingEherkenning
                    ? 'Bezig met inloggen...'
                    : 'Inloggen met IRMA'}
                </span>
              </a>
            </p>
            <Heading size="tiny" el="h4">
              Hebt u nog geen IRMA? Regel dit dan eerst.
            </Heading>
            <p>
              Ga naar{' '}
              <a rel="noopener noreferrer" href="https://irma.app">
                IRMA: een nieuwe manier van inloggen
              </a>{' '}
              voor meer informatie.
            </p>
          </div>
        )}
        {FeatureToggle.eherkenningActive && (
          <div
            className={classnames(
              styles.LoginOption,
              styles['LoginOption--eherkenning']
            )}
          >
            <Heading className={styles.LoginOptionHeading} size="tiny" el="h3">
              Voor ondernemers
            </Heading>
            <p>
              <a
                ref={loginButton}
                role="button"
                href={LOGIN_URL_EHERKENNING}
                onClick={() => setRedirectingEherkenning(true)}
                rel="noopener noreferrer"
                className={classnames(
                  styles.LoginBtn,
                  styles['LoginBtn--eherkenning'],
                  (isRedirecting || isRedirectingEherkenning) &&
                    styles.LoginBtnDisabled
                )}
              >
                <span className={styles.LoginLogoWrap}>
                  <img
                    src={EherkenningLogo}
                    alt="eHerkenning logo"
                    className={styles.LoginLogo}
                  />
                </span>
                <span className={styles.LoginButtonText}>
                  {isRedirectingEherkenning
                    ? 'Bezig met inloggen...'
                    : 'Inloggen met EHerkenning'}
                </span>
              </a>
            </p>
            <Heading size="tiny" el="h4">
              U hebt EHerkenning niveau {IS_PRODUCTION ? '2+' : '3'} nodig om in
              te loggen.
            </Heading>
            <p>
              Ga naar{' '}
              <a rel="noopener noreferrer" href="https://eherkenning.nl">
                eherkenning.nl
              </a>{' '}
              voor meer informatie.
            </p>
          </div>
        )}
        <Heading size="tiny" el="h4">
          Vragen over Mijn Amsterdam?
        </Heading>
        <p className={styles.FaqInfo}>
          Kijk bij{' '}
          <LinkdInline
            external={true}
            href={ExternalUrls.MIJN_AMSTERDAM_VEELGEVRAAGD}
          >
            veelgestelde vragen over Mijn Amsterdam
          </LinkdInline>
        </p>
      </PageContent>
    </TextPage>
  );
}
