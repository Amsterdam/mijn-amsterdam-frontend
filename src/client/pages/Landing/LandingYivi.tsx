import classnames from 'classnames';
import { useRef, useState } from 'react';
import {
  Heading,
  LinkdInline,
  MaintenanceNotifications,
  PageContent,
  PageHeading,
  TextPage,
} from '../../components';
import { LOGIN_URL_YIVI } from '../../config/api';
import styles from './Landing.module.scss';
import { ReactComponent as YiviLogo } from './yivi-logo.svg';

export default function Landing() {
  const loginButton = useRef(null);
  const [isRedirectingYivi, setRedirectingYivi] = useState(false);
  const isRedirectingAny = isRedirectingYivi;

  return (
    <TextPage>
      <PageHeading className={styles.Heading}>
        Melding openbare ruimte volgen
      </PageHeading>
      <PageContent className={styles.LandingContent} id="skip-to-id-AppContent">
        <p>
          U heeft een melding openbare ruimte gedaan. En u wilt weten wat er met
          uw melding gebeurt. In Mijn Amsterdam kunt u uw melding bekijken. Dit
          is een proef van de gemeente Amsterdam.
        </p>
        <MaintenanceNotifications fromApiDirectly={true} page="yivisignalen" />
        <div
          className={classnames(
            styles.LoginOption,
            styles['LoginOption--yivi']
          )}
        >
          <Heading className={styles.LoginOptionHeading} size="large" el="h3">
            Inloggen met Yivi
          </Heading>
          <p>
            Wij maken voor deze proef gebruik van de app Yivi. U kunt met Yivi
            inloggen zonder onnodige gegevens met ons te delen. Het enige dat
            nodig is, is het e-mailadres dat u heeft opgegeven bij het doen van
            de melding.
          </p>
          <p>
            <a
              ref={loginButton}
              role="button"
              href={LOGIN_URL_YIVI}
              onClick={() => setRedirectingYivi(true)}
              rel="noopener noreferrer"
              className={classnames(
                styles.LoginBtn,
                styles['LoginBtn--yivi'],
                isRedirectingAny && styles.LoginBtnDisabled
              )}
            >
              <span className={styles.YiviLogoWrap}>
                <YiviLogo />
              </span>
              <span className={styles.LoginButtonText}>
                {isRedirectingYivi
                  ? 'Bezig met inloggen...'
                  : 'Inloggen met YIVI'}
              </span>
            </a>
          </p>
          <p>
            Kijk voor meer informatie over de proef en Yivi bij{' '}
            <LinkdInline
              external
              href="https://www.amsterdam.nl/veelgevraagd/?caseid=%7Bfbeab9ad-81e4-4fee-9dfc-3fd62eb92719%7D"
            >
              Melding openbare ruimte volgen
            </LinkdInline>
            .
          </p>
          <p>
            Login met DigiD voor de volledige versie van{' '}
            <LinkdInline external href={process.env.BFF_FRONTEND_URL}>
              Mijn Amsterdam
            </LinkdInline>
            .
          </p>
          <Heading size="medium" el="h4">
            Download Yivi
          </Heading>
          <p>
            <LinkdInline href="https://www.yivi.app/download" external>
              Download gratis de Yivi-app
            </LinkdInline>
          </p>
        </div>

        <Heading size="medium" el="h4">
          Vragen over Mijn Amsterdam?
        </Heading>
        <p className={styles.FaqInfo}>
          Kijk bij{' '}
          <LinkdInline
            external
            href="https://www.amsterdam.nl/veelgevraagd/?productid=%7B68422ECA-8C56-43EC-A9AA-B3DF190B5077%7D"
          >
            veelgestelde vragen over Mijn Amsterdam
          </LinkdInline>
        </p>
      </PageContent>
    </TextPage>
  );
}
