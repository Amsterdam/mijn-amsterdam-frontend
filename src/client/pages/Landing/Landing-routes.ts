import { LandingPage } from './Landing.tsx';

export const LandingRoute = {
  route: '/',
  Component: LandingPage,
  props: { index: true },
  public: true,
  private: false,
};

export const LandingRoutes = [LandingRoute];

export const DASHBOARD_PAGE_DOCUMENT_TITLE = 'Inloggen | Mijn Amsterdam';
