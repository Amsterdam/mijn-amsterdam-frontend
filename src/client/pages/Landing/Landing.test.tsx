import { shallow } from 'enzyme';
import React from 'react';
import Landing from './Landing';
import { RecoilRoot } from 'recoil';
import { sessionAtom, SessionState } from '../../hooks/api/useSessionApi';

const appState = { isAuthenticated: false } as SessionState;

describe('<Landing />', () => {
  it('Renders without crashing', () => {
    expect(
      shallow(
        <RecoilRoot
          initializeState={snapshot => snapshot.set(sessionAtom, appState)}
        >
          <Landing />
        </RecoilRoot>
      ).html()
    ).toMatchSnapshot();
  });
});
