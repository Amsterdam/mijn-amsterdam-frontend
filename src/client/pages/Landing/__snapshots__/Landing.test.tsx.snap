// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Landing /> Renders without crashing 1`] = `
<DocumentFragment>
  <main
    class="Page TextPage"
  >
    <header
      class="PageHeading Heading"
    >
      <div
        class="HeadingInner"
      >
        <h2
          class="Heading Heading__large"
        >
          Welkom op Mijn Amsterdam
        </h2>
      </div>
    </header>
    <div
      class="PageContent LandingContent"
      id="skip-to-id-AppContent"
    >
      <p>
        Mijn Amsterdam is uw persoonlijke online pagina bij de gemeente Amsterdam.
      </p>
      <div
        class="LoginOption"
      >
        <h3
          class="Heading Heading__tiny LoginOptionHeading"
        >
          Voor particulieren en eenmanszaken
        </h3>
        <p>
          <a
            class="LoginBtn"
            href="/api/v1/auth/digid/login"
            rel="noopener noreferrer"
            role="button"
          >
            <span
              class="LoginLogoWrap"
            >
              <svg
                viewBox="0 0 150 150"
                xmlns="http://www.w3.org/2000/svg"
              >
                <title>
                  DigiD logo
                </title>
                <path
                  d="M136 150H14c-8 0-14-6-14-14V14C0 6 6 0 14 0h122c8 0 14 6 14 14v122c0 8-6 14-14 14z"
                />
                <path
                  d="M17 115V79h10c12 0 19 6 19 17 0 13-8 19-19 19H17zm6-6h4c7 0 12-4 12-13 0-8-5-12-13-12h-3v25zM54 77c3 0 4 1 4 3s-1 4-4 4c-2 0-3-2-3-4s1-3 3-3zm3 38h-6V88h6v27zM72 109h6c6 0 9 3 9 7 0 5-4 9-14 9-8 0-11-2-11-7 0-2 1-4 4-6l-2-3c0-2 1-3 3-4-3-2-4-4-4-8 0-6 4-10 11-10l4 1h9v4h-4l2 5c0 6-4 9-12 9h-3l-1 1c0 2 1 2 3 2zm1 12c6 0 8-2 8-4s-1-2-3-2l-9-1-2 3c0 2 2 4 6 4zm6-24c0-3-2-5-5-5s-5 1-5 5 1 5 5 5c3 0 5-1 5-5z"
                  fill="#fff"
                />
                <path
                  d="M94 77c2 0 3 1 3 3s-1 4-3 4c-3 0-4-2-4-4s1-3 4-3zm3 38h-6V88h6v27zM105 115V79h10c12 0 18 6 18 17 0 13-7 19-19 19h-9zm6-6h4c7 0 12-4 12-13 0-8-5-12-13-12h-3v25z"
                  fill="#E17000"
                />
              </svg>
            </span>
            <span
              class="LoginButtonText"
            >
              Inloggen met DigiD
            </span>
          </a>
        </p>
        <h4
          class="Heading Heading__tiny"
        >
          Hebt u nog geen DigiD? Regel dit dan eerst.
        </h4>
        <p>
          Ga naar 
          <a
            href="https://www.digid.nl/aanvragen"
            rel="noopener noreferrer"
          >
            DigiD aanvragen
          </a>
        </p>
      </div>
      <div
        class="LoginOption LoginOption--eherkenning"
      >
        <h3
          class="Heading Heading__tiny LoginOptionHeading"
        >
          Voor ondernemers
        </h3>
        <p>
          <a
            class="LoginBtn LoginBtn--eherkenning"
            href="/api/v1/auth/eherkenning/login"
            rel="noopener noreferrer"
            role="button"
          >
            <span
              class="LoginLogoWrap"
            >
              <svg
                height="28px"
                version="1.1"
                viewBox="0 0 46 28"
                width="46px"
                xmlns="http://www.w3.org/2000/svg"
              >
                <title>
                  eHerkenning logo
                </title>
                <g
                  fill="#DF0067"
                  transform="translate(25.134021, 0.000000)"
                >
                  <rect
                    height="11.2865979"
                    width="4.45773196"
                    x="0.0948453608"
                    y="15.7443299"
                  />
                  <rect
                    height="11.0020619"
                    width="4.45773196"
                    x="0.0948453608"
                    y="0.331958763"
                  />
                  <polygon
                    points="16.3134021 0.331958763 16.3134021 11.3814433 6.7814433 11.3814433 4.8371134 15.6969072 16.3134021 15.6969072 16.3134021 27.0309278 20.8185567 27.0309278 20.8185567 0.331958763"
                  />
                </g>
                <path
                  d="M7.72989691,27.0309278 C3.69896907,27.0309278 0.379381443,23.7113402 0.379381443,19.6804124 L0.379381443,19.6804124 L0.379381443,7.72989691 C0.379381443,3.65154639 3.65154639,0.379381443 7.72989691,0.379381443 L7.72989691,0.379381443 L21.1505155,0.379381443 L21.1505155,4.97938144 L7.72989691,4.97938144 C6.21237113,4.97938144 4.97938144,6.21237113 4.97938144,7.77731959 L4.97938144,7.77731959 L4.97938144,19.7278351 C4.97938144,21.2453608 6.21237113,22.5257732 7.72989691,22.5257732 L7.72989691,22.5257732 L21.1505155,22.5257732 L21.1505155,27.0309278 L7.72989691,27.0309278 L7.72989691,27.0309278 Z"
                  fill="#133370"
                />
                <polygon
                  fill="#133370"
                  points="7.91958763 11.3814433 7.91958763 15.6969072 18.2103093 15.6969072 20.1546392 11.3814433"
                />
              </svg>
            </span>
            <span
              class="LoginButtonText"
            >
              Inloggen met eHerkenning
            </span>
          </a>
        </p>
        <h4
          class="Heading Heading__tiny"
        >
          U hebt eHerkenning niveau 3 nodig om in te loggen.
        </h4>
        <p>
          Ga naar 
          <a
            href="https://eherkenning.nl"
            rel="noopener noreferrer"
          >
            eherkenning.nl
          </a>
           voor meer informatie.
        </p>
      </div>
      <h4
        class="Heading Heading__tiny"
      >
        Vragen over Mijn Amsterdam?
      </h4>
      <p
        class="FaqInfo"
      >
        Kijk bij 
        <a
          class="Button Button__inline Button__lean Linkd LinkedInline"
          href="https://www.amsterdam.nl/veelgevraagd/?productid={68422ECA-8C56-43EC-A9AA-B3DF190B5077}"
          rel="external noopener noreferrer"
        >
          veelgestelde vragen over Mijn Amsterdam
        </a>
      </p>
    </div>
  </main>
</DocumentFragment>
`;
